src/features/core/metrics/errorReporting/providers/Bugsnag.js

```diff
@@ -0,0 +1,240 @@
+/* eslint-disable no-underscore-dangle */
+import ScriptLoader from '../../../../../utils/ScriptLoader';
+import { configToIntNames } from '../../../../../utils/config_to_integration_names';
+import {
+  ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME,
+  MAX_WAIT_FOR_INTEGRATION_LOAD,
+} from '../../../../../utils/constants';
+import { get } from '../../../../../utils/utils';
+
+// Using the Bugsnag integration version to avoid version issues
+const BUGSNAG_CDN_URL = 'https://d2wy8f7a9ursnm.cloudfront.net/v6/bugsnag.min.js';
+const BUGSNAG_VALID_MAJOR_VERSION = '6';
+const ERROR_REPORT_PROVIDER_NAME_BUGSNAG = 'rs-bugsnag';
+const BUGSNAG_LIB_INSTANCE_GLOBAL_KEY_NAME = 'bugsnag'; // For version 7 this is 'Bugsnag'
+const GLOBAL_LIBRARY_OBJECT_NAMES = [
+  ERROR_REPORT_PROVIDER_NAME_BUGSNAG,
+  'Bugsnag',
+  BUGSNAG_LIB_INSTANCE_GLOBAL_KEY_NAME,
+  ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME,
```

Comment:
This is also checked separately at the initial stages.

Reply:
need more time to check this one as it has been a while